pkg/kf/commands/install/gke.go

```diff
@@ -0,0 +1,908 @@
+	}
+
+	if idx == 0 {
+		// Create new cluster
+		return createNewCluster(ctx, projID)
+	}
+
+	// Use existing cluster
+	// We need to figure out the zone
+	zone, err := clusterZone(ctx, projID, clusterName)
+	return clusterName, zone, err
+}
+
+func randName(prefix string, args ...interface{}) string {
+	return fmt.Sprintf(prefix, args...) + strconv.FormatInt(time.Now().UnixNano(), 36)
+}
+
+var (
+	nameRegexp = regexp.MustCompile(`^[a-z][0-9a-zA-Z-]{5,29}$`)
+	// hostnameRegexp is from https://stackoverflow.com/a/106223
+	hostnameRegexp = regexp.MustCompile(`^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])$`)
+)
+
+func createProject(ctx context.Context) (string, error) {
+	ctx = setLogPrefix(ctx, "New Project")
+
+	for {
+		name, err := namePrompt(
+			ctx,
+			labelColor.Sprint("Project Name (not ID): "),
+			randName("kf-"),
+		)
+		if err != nil {
+			return "", err
+		}
+
+		Logf(ctx, "creating new project %s", name)
+		_, err = gcloud(ctx, "-q", "projects", "create", "--name", name)
+		if err != nil {
+			return "", err
+		}
+
+		Logf(ctx, "fetching project ID for %s", name)
+		projects, err := gcloud(
+			ctx,
+			"projects",
+			"list",
+			"--filter", fmt.Sprintf("name:%q", name),
+			"--format", "value(projectId)",
+		)
+
+		if err != nil {
+			return "", err
+		}
+
+		if len(projects) != 1 {
+			return "", fmt.Errorf(
+				"something went wrong while trying to fetch the project ID for %s: %s",
+				name,
+				strings.Join(projects, "\n"),
+			)
+		}
+
+		projID := projects[0]
+		Logf(ctx, "Created project %s with a project ID %s", name, projID)
+		return projID, nil
+	}
+}
+
+func hostnamePrompt(ctx context.Context, label, def string) (string, error) {
+	prompt := promptui.Prompt{
+		Label: labelColor.Sprint(label),
+		Validate: func(input string) error {
+			if !hostnameRegexp.MatchString(input) {
+				return errors.New("invalid hostname")
+			}
+			return nil
+		},
+		Default: def,
+	}
+
+	name, err := prompt.Run()
+	if err != nil {
+		return "", err
+	}
+
+	return name, nil
+}
+
+func namePrompt(ctx context.Context, label, def string) (string, error) {
+	prompt := promptui.Prompt{
+		Label: labelColor.Sprint(label),
+		Validate: func(input string) error {
+			if !nameRegexp.MatchString(input) {
+				return errors.New("invalid name")
+			}
+			return nil
+		},
+		Default: def,
+	}
+
+	name, err := prompt.Run()
+	if err != nil {
+		return "", err
+	}
+
+	return name, nil
+}
+
+func selectPrompt(
+	ctx context.Context,
+	label string,
+	items ...string,
+) (int, string, error) {
+	p := promptui.Select{
+		Label:             labelColor.Sprint(label),
+		StartInSearchMode: true,
+		Searcher:          searcher(items),
+		Items:             items,
+	}
+	return p.Run()
+}
+
+func selectYesNo(ctx context.Context, label string) (bool, error) {
+	idx, _, err := selectPrompt(ctx, label, "yes", "no")
+	if err != nil {
+		return false, err
+	}
+
+	return idx == 0, nil
```

Comment:
I'd prefer a string check to an index check in case the APIs or ordering changes in the future.